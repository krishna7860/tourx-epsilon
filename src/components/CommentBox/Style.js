import styled from "styled-components";
import TextareaAutosize from "@material-ui/core/TextareaAutosize";

export const TextAreaSection = styled(TextareaAutosize)`
  border: none;
  overflow: auto;
  outline: none;
  width: 100%;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
  box-shadow: none;
  resize: none;
`;
export const Wrapper = styled.div`
  width: 80%;
  margin: 20px 0px;
  padding: 20px;
  border: 1px solid #d4d4d4;
  border-radius: 5px;
  @media screen and (max-width: 768px) {
    width: 100%;
  }
`;
export const ActionSection = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  cursor: pointer;
`;

export const ErrorSpan = styled.div`
  font-size: 18px;
  color: red;
`;
