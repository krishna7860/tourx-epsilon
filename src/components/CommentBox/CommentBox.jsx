/* eslint-disable react/prop-types */
import React from "react";
import SendIcon from "@material-ui/icons/Send";

import { TextAreaSection, Wrapper, ActionSection, ErrorSpan } from "./Style";

const CommentBox = (props) => {
  const { isAuth, handleAddReview } = props;
  const [state, setstate] = React.useState("");
  return (
    <Wrapper>
      <TextAreaSection
        aria-label="minimum height"
        rowsMin={3}
        placeholder="Enter comments"
        disabled={!isAuth}
        value={state}
        onChange={(e) => setstate(e.target.value)}
      />
      <ErrorSpan>
        {!isAuth ? "Please Login First to add the Review" : ""}
      </ErrorSpan>
      <ActionSection>
        <SendIcon
          style={{ cursor: "pointer" }}
          color="primary"
          onClick={() => {
            handleAddReview(state);
            setstate("");
          }}
        />
      </ActionSection>
    </Wrapper>
  );
};

export default CommentBox;
