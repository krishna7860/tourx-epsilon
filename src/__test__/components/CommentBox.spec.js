import React from "react";
import { render } from "@testing-library/react";
import CommentBox from "../../components/CommentBox/CommentBox";

describe("components/commentBox", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(<CommentBox />);
    expect(asFragment()).toMatchSnapshot();
  });
});
