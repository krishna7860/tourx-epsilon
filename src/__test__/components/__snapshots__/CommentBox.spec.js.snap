// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commentBox should be rendered properly 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT kzlWAZ"
  >
    <textarea
      aria-label="minimum height"
      class="sc-bdnxRM lcHJYh"
      placeholder="Enter comments"
      rows="3"
      style="height: -4px; overflow: hidden;"
    />
    <textarea
      aria-hidden="true"
      class="sc-bdnxRM lcHJYh"
      readonly=""
      style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: auto;"
      tabindex="-1"
    />
    <div
      class="sc-dlnjwi hEuURr"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
